lib/TextRoles/SpecialTextRole.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\RST\TextRoles;
+
+use Doctrine\RST\Span\SpanProcessor;
+use Doctrine\RST\Span\SpanToken;
+
+use function preg_replace_callback;
+
+class SpecialTextRole extends WrapperTextRole
+{
+    /*
+     * Does this text role have a special syntax like ``*cursive*``?
+     */
+    public function hasSpecialSyntax(): bool
+    {
+        return true;
+    }
+
+    public function replaceTokens(SpanProcessor $spanProcessor, string $span, string $pattern): string
+    {
+        $span = (string) preg_replace_callback(
+            $pattern,
```

Comment:
Can `$pattern` be supplied by a user of this library? If yes, I'd expect `preg_last_error()` to be called at some point.

Reply:
I think it should be `protected` as only implementing text roles can use this